allennlp/models/coreference_resolution/coref.py

```diff
@@ -361,12 +412,14 @@ def make_output_human_readable(self, output_dict: Dict[str, torch.Tensor]):
         # A tensor of shape (num_spans_to_keep, max_antecedents), representing the indices
         # of the predicted antecedents with respect to the 2nd dimension of `batch_top_spans`
         # for each antecedent we considered.
-        antecedent_indices = output_dict["antecedent_indices"].detach().cpu()
+        batch_antecedent_indices = output_dict["antecedent_indices"].detach().cpu()
```

Comment:
Ah, I think I understand why we need the batch dim - you want to have different possible numbers of antecedent indices per batch?

Reply:
Not just different numbers of antecedent indices, but really different values of antecedent indices, due to the flexibility of the coarse-to-fine scorer.